tests/template/template_cleantest_test.go

```diff
@@ -35,98 +35,53 @@ var _ = Describe("Example of a clean test", func() {
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 		project = helper.CreateRandProject()
 
+		// we will be testing components that are created from the current directory
+		// switch to the clean context dir before each test
+		originalDir = helper.Getwd()
+		helper.Chdir(context)
+
 	})
 
 	AfterEach(func() {
 		helper.DeleteProject(project)
+		// go back to original directory after each test
+		helper.Chdir(originalDir)
 		helper.DeleteDir(context)
 		os.Unsetenv("GLOBALODOCONFIG")
 	})
 
-	var _ = Context("when component is in the current directory", func() {
-
-		// we will be testing components that are created from the current directory
-		// switch to the clean context dir before each test
-		var _ = JustBeforeEach(func() {
-			originalDir = helper.Getwd()
-			helper.Chdir(context)
+	Context("when project from KUBECONFIG is used", func() {
+		// Set active project for each test spec
+		JustBeforeEach(func() {
+			originalProject = oc.GetCurrentProject()
+			// WARNING: this project switching makes it impossible to run this in parallel
+			// it should set different KUBECONFIG before witching project
+			oc.SwitchProject(project)
 		})
 
-		// go back to original directory after each test
-		var _ = JustAfterEach(func() {
-			helper.Chdir(originalDir)
-		})
-
-		var _ = Context("when project from KUBECONFIG is used", func() {
-			// Set active project for each test spec
-			JustBeforeEach(func() {
-				originalProject = oc.GetCurrentProject()
-				// WARNING: this project switching makes it impossible to run this in parallel
-				// it should set different KUBECONFIG before witching project
-				oc.SwitchProject(project)
-			})
-			// go back to original project after each test
-			JustAfterEach(func() {
-				oc.SwitchProject(originalProject)
-			})
-
-			It("create local nodejs component and push code", func() {
-				helper.CopyExample(filepath.Join("source", "nodejs"), context)
-
-				//helper.CmdShouldPass("odo component create nodejs")
-				helper.CmdShouldPass("odo", "component", "create", "nodejs")
-				//TODO: verify that config was properly created
-				helper.CmdShouldPass("odo", "push")
-				//TODO: verify resources on cluster
-			})
-
+		// go back to original project after each test
+		JustAfterEach(func() {
+			oc.SwitchProject(originalProject)
```

Comment:
Now it is no where used in the test suite. It has been replaced with ```--project``` flag. So can remove it completely. And try to replicate the template as per our current test suite use.

Reply:
I can see we are still using this here - https://github.com/openshift/odo/blob/master/tests/integration/component.go#L67-L75 and some more places also. So do you mean we should remove from those places also or how is it

Reply:
@amitkrout Yes, I don't feel any significance of `oc.SwitchProject(originalProject)` in our test script. So I removed it from template but as we are still using it at some places lets remove this from there also.